tests/test_operator.py

```diff
@@ -980,6 +980,36 @@ def test_equations_mixed_timedim_stepdim(self):
         assert trees[0][-1].nodes[1].write == u2
 
 
+@skipif_yask
+class TestRegions(object):
+
+    def test_domain_vs_interior(self):
+        """
+        Tests regions work properly in terms of code generation and runtime
+        argument derivation.
+        """
+        grid = Grid(shape=(4, 4, 4))
+        x, y, z = grid.dimensions
+        t = grid.stepping_dim  # noqa
+
+        u = TimeFunction(name='u', grid=grid)  # noqa
+        eqs = [Eq(u.forward, u + 1),
+               Eq(u.forward, u.forward + 2, region=INTERIOR)]
```

Comment:
Weird place but looking into it, how easy would it be to create a 'BOUNDARY' region that would be 'DOMAIN' - 'INTERIOR'

Reply:
well, doable, but isn't it more natural to have equations with fixed array indices to update the boundary ?

Reply:
So this one is tricky, indices are tricky to manage for boundary for non 1D problems. This is not for this PR, but I think a nice way to have "automatic" indexing of the boundary would help a lot for 2D/3D.

Reply:
because you have an explosion in the number of equations?

Reply:
Yes, and also the iteration space becomes quite tricky to write by hand for non specialist.

Reply:
Lets design BOUNDARY together next month